src/main/java/com/google/sps/servlets/QueryServlet.java

```diff
@@ -67,8 +72,14 @@ public void doPost(HttpServletRequest servletRequest, HttpServletResponse servle
         String radius = servletRequest.getParameter("radius");
         String type = "restaurant";
         String searchTerms = servletRequest.getParameter("searchTerms");
-        String urlStr = "https://maps.googleapis.com/maps/api/place/nearbysearch/json?location=" + lat + "," + lon + "&radius=" + radius + "&type=" + type + "&keyword=" + searchTerms + "&key=" + apiKey;
+        searchTerms = searchTerms.replaceAll("\\s", "+");
```

Comment:
You can use [URLEncoder](https://docs.oracle.com/en/java/javase/14/docs/api/java.base/java/net/URLEncoder.html) to do this.

Reply:
done